lib/trento/domain/sap_system/instance.ex

```diff
@@ -3,25 +3,19 @@ defmodule Trento.Domain.SapSystem.Instance do
   This module represents a SAP System instance.
   """
 
-  alias Trento.Domain.Enums.Health
+  require Trento.Domain.Enums.Health, as: Health
 
-  defstruct [
-    :sid,
-    :instance_number,
-    :features,
-    :host_id,
-    :health,
-    :system_replication,
-    :system_replication_status
-  ]
+  @required_fields []
 
-  @type t :: %__MODULE__{
-          sid: String.t(),
-          instance_number: String.t(),
-          features: String.t(),
-          host_id: String.t(),
-          health: Health.t(),
-          system_replication: String.t(),
-          system_replication_status: String.t()
-        }
+  use Trento.Type
+
+  deftype do
+    field :sid, :string
+    field :instance_number, :integer
```

Comment:
I'm not sure if we should change the instance number to integer. Yes it is an integer, but we always use it as a 2 digit string. I would be conservative with this an keep string

Reply:
mmh i think i mistyped it thank you